kong/db/schema/init.lua

```diff
@@ -1736,25 +1760,25 @@ function Schema:process_auto_fields(data, context, nulls, opts)
       local value = data[sname]
       if value ~= nil then
         local _, err = self:validate_field(sdata, value)
+        local _, deprecation_error = validate_deprecation_exclusiveness(data, value, sname, sdata)
         if err then
           errs[sname] = err
           has_errs = true
+        elseif deprecation_error then
+          errs[sname] = deprecation_error
+          has_errs = true
         else
           data[sname] = nil
           local new_values = sdata.func(value)
           if new_values then
-            -- a shorthand field may have a deprecation property, that is used
-            -- to determine whether the shorthand's return value takes precedence
-            -- over the similarly named actual schema fields' value when both
-            -- are present. On deprecated shorthand fields the actual schema
-            -- field value takes the precedence, otherwise the shorthand's
-            -- return value takes the precedence.
-            local deprecation = sdata.deprecation
             for k, v in pairs(new_values) do
```

Comment:
Do we consider it a breaking change? As we no longer have this deprecated value precedence behavior (it seems if multiple value involved it will merge both)

Reply:
I reversed those changes.